distribution/Dockerfile

```diff
@@ -50,18 +50,33 @@ EXPOSE ${agent_port}
 # The only dependencies for Jenkins Evergreen are:
 #   * supervisor
 #   * nodejs
-RUN apk add --no-cache git \
+RUN yum update -y
+RUN yum install -y --setopt=skip_missing_names_on_install=False \
+        epel-release
+RUN yum install -y --setopt=skip_missing_names_on_install=False \
+                        git \
                         ca-certificates \
-                        openssh-client \
+                        openssh-clients \
                         unzip \
                         bash \
-                        openjdk8 \
+                        java-1.8.0-openjdk \
                         supervisor \
-                        ttf-dejavu \
+                        dejavu-sans-mono-fonts \
                         curl \
                         socat \
                         nginx
 
+# Use the auto-install script, but prevent any unnoticed change
+# https://linuxize.com/post/how-to-install-node-js-on-centos-7/
+RUN curl -sL https://rpm.nodesource.com/setup_10.x > /tmp/install-node && \
+    echo "2050d9584f54fd1da7dfc54f52eac24d  /tmp/install-node" | md5sum --check && \
+    cat /tmp/install-node| bash - && \
+    rm /tmp/install-node
+RUN yum install -y --setopt=skip_missing_names_on_install=False \
+    nodejs
+RUN node --version | grep '^v10.' > /dev/null && \
```

Comment:
I will spend a bit of time later to optimize layers, I could probably save some disk space and download time here.

Reply:
Also about the `grep`, there are obviously not needed in terms of image, but it saved a lot of time, and I feel is good to catch it early in the future.

We have tests in `offline-tests.sh` that check that too, so in general these tests are _almost_ moot now. Though I feel like they are still interesting for when/if we create new flavors and we don't handle node/npm versions in new ``Dockerfile``s